### PR TITLE
Jshook/talk improvements

### PR DESCRIPTION
**Pull request by jshook**
Various improvements I made to 5.21 so that it works as expected. These need to be documented.

### CODE REVIEW

File: nb-api/src/main/java/io/nosqlbench/nb/api/components/core/NBMetricsBuffer.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class NBMetricsBuffer {
+
+    private final static Logger logger = LogManager.getLogger(NBMetricsBuffer.class);
+    private List<NBMetric> summaryMetrics = new ArrayList<>();
+//    private PrintStream out = System.out;
```
Comment (MarkWolters) [inline nb-api/src/main/java/io/nosqlbench/nb/api/components/core/NBMetricsBuffer.java:36]: dead code

File: nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ParameterMap.java [hunk 2]
```diff
@@ -181,9 +181,9 @@ public void setSilently(String paramName, Object newValue) {
 
 
     public void set(String paramName, Object newValue) {
-        if (paramName.equals("cycles")) {
-            logger.warn("Setting 'cycles' on the parameter map is likely causing a bug in your activity. Call setCycles on the def instead.");
-        }
+//        if (paramName.equals("cycles")) {
+//            logger.warn("Setting 'cycles' on the parameter map is likely causing a bug in your activity. Call setCycles on the def instead.");
```
Comment (MarkWolters) [inline nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ParameterMap.java:185]: dead code

**Review by MarkWolters — APPROVED** [review]
LGTM!

File: nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/CyclesSpec.java [hunk 3]
```diff
@@ -28,9 +28,9 @@ public record CyclesSpec(long first_inclusive, long last_exclusive, String first
         if (first_inclusive>last_exclusive) {
             throw new InvalidParameterException("cycles must start with a lower first cycle than last cycle");
         }
-        if (first_inclusive==last_exclusive) {
-            logger.warn("This cycles interval means zero total:" + this);
-        }
+//        if (first_inclusive==last_exclusive) {
```
[the remark this answers is not in the document]
Comment (jshook) [reply nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/CyclesSpec.java:31]: fixing

File: nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ParameterMap.java (hunk 2 above)
Comment (jshook) [reply nb-api/src/main/java/io/nosqlbench/nb/api/engine/activityimpl/ParameterMap.java:185]: fixing

**Pull request merged**